### PR TITLE
Adds type checks via `pyright` to CI

### PR DESCRIPTION
**Pull request by ikamensh**
Contribution towards #2452
Adds type checks via `pyright` - first step for having typing in gym, and two first typechecked files (version.py and logging.py).

While `mypy` can be considered official / reference type checker, in practice it has difficulties supporting all accepted typing PEPs without workarounds in user code. As mentioned in the discussion in #2452, I have much better experience in production with `pyright` - it's signal to noise ratio is much higher. Most reported issues are actual deficiencies / missing typing. 

Advantages:
1) Pyright is well maintained, few open issues.
2) Developers in Visual Studio Code can have SAME type checking right in their IDE.

Known drawbacks:
1) I couldn't find a working way of including `pyright` in pre-commit.

### CODE REVIEW

File: gym/logger.py [hunk 1]
```diff
@@ -9,42 +10,47 @@
 ERROR = 40
 DISABLED = 50
 
-MIN_LEVEL = 30
```
Comment (ikamensh) [inline gym/logger.py:12]: type checker: modifying a constant (because fo SNAKE_CASE)

**Comment by jkterry1** [comment]
This looks good, but can you please merge the new .md file you propose into CONTRIBUTING.md?

**Comment by ikamensh** [comment]
done

**Comment by jkterry1** [comment]
Thanks a ton for all your PRs!

Could you please fix the merge conflict? It looks like merging your other PRs resulted in one

**Comment by ikamensh** [comment]
You're welcome! :) Conflict + CI should be fixed now

**Pull request merged**